File: LC/3100/3100.py
# link of the question: https://leetcode.com/problems/water-bottles-ii/

class Solution:
    def maxBottlesDrunk(self, numBottles: int, numExchange: int) -> int:
        total_drink = numBottles
        empty_bottle = numBottles
        while empty_bottle >= numExchange:
            empty_bottle = empty_bottle - numExchange + 1
            numExchange += 1
            total_drink += 1
        return total_drink

File: online_judge/LC/3100/3100.cpp
// link of the question: https://leetcode.com/problems/water-bottles-ii/

class Solution {
public:
    int maxBottlesDrunk(int numBottles, int numExchange) {
        int total_drink = numBottles;
        int empty_bottle = numBottles;

        while (empty_bottle >= numExchange) {
            empty_bottle = empty_bottle - numExchange + 1;
            numExchange++;
            total_drink++;
        }

        return total_drink;
    }
};